src/dataframerow/dataframerow.jl

```diff
@@ -1,85 +1,185 @@
+    remap = (1:last(cols)) .- first(cols) .+ 1
+    DataFrameRow(df, row, cols, remap)
+end
+
+@inline function DataFrameRow(df::AbstractDataFrame, row::Integer, ::Colon)
+    @boundscheck if !checkindex(Bool, axes(df, 1), row)
+        throw(BoundsError("attempt to access a data frame with $(nrow(df)) " *
+                          "rows at index $row"))
+    end
+    cols = axes(df, 2)
+    DataFrameRow(df, row, cols, cols)
+end
+
+@inline DataFrameRow(df::AbstractDataFrame, row::Integer, cols::AbstractVector{Int}) =
+    DataFrameRow(df, row, convert(Vector{Int}, cols))
+@inline DataFrameRow(df::AbstractDataFrame, row::Integer, cols::AbstractUnitRange{Int}) =
+    DataFrameRow(df, row, convert(UnitRange{Int}, cols))
+@inline DataFrameRow(df::AbstractDataFrame, row::Integer, cols) =
+    DataFrameRow(df, row, index(df)[cols])
+@inline DataFrameRow(df::AbstractDataFrame, row::Integer, cols::ColumnIndex) =
+    throw(ArgumentError("invalid column vector $cols"))
 
-Base.parent(r::DataFrameRow) = getfield(r, :df)
-Base.parentindices(r::DataFrameRow) = (row(r), axes(parent(r), 2))
 row(r::DataFrameRow) = getfield(r, :row)
+Base.parent(r::DataFrameRow) = getfield(r, :df)
+Base.parentindices(r::DataFrameRow) = (row(r), getfield(r, :cols))
 
 Base.view(adf::AbstractDataFrame, rowind::Integer, ::Colon) =
-    DataFrameRow(adf, rowind)
+    DataFrameRow(adf, rowind, :)
 Base.view(sdf::SubDataFrame, rowind::Integer, ::Colon) =
-    DataFrameRow(parent(sdf), rows(sdf)[rowind])
+    DataFrameRow(parent(sdf), rows(sdf)[rowind], parentindices(sdf)[2])
 
-# Here a corner case is when colinds=[] and we pass a valid rowind
-# into adf. We will throw an error in this case.
-# The consequence is that it is impossible to create a DataFrameRow without columns.
 Base.view(adf::AbstractDataFrame, rowind::Integer, colinds::AbstractVector) =
-    DataFrameRow(adf[colinds], rowind)
+    DataFrameRow(adf, rowind, colinds)
 Base.view(sdf::SubDataFrame, rowind::Integer, colinds::AbstractVector) =
-    DataFrameRow(parent(sdf)[colinds], rows(sdf)[rowind])
+    DataFrameRow(parent(sdf), rows(sdf)[rowind], index(sdf)[colinds])
 
 # Same here. It is impossible to create a DataFrameRow without columns.
 Base.getindex(df::DataFrame, rowind::Integer, colinds::AbstractVector) =
-    DataFrameRow(df[colinds], rowind)
+    DataFrameRow(df, rowind, colinds)
 Base.getindex(sdf::SubDataFrame, rowind::Integer, colinds::AbstractVector) =
-    DataFrameRow(parent(sdf)[colinds], rows(sdf)[rowind])
+    DataFrameRow(parent(sdf), rows(sdf)[rowind], index(sdf)[colinds])
 Base.getindex(df::DataFrame, rowind::Integer, ::Colon) =
-    DataFrameRow(df, rowind)
+    DataFrameRow(df, rowind, :)
 Base.getindex(sdf::SubDataFrame, rowind::Integer, ::Colon) =
-    DataFrameRow(parent(sdf), rows(sdf)[rowind])
+    DataFrameRow(parent(sdf), rows(sdf)[rowind], parentindices(sdf)[2])
+
+getparentcols(r::DataFrameRow, idx::Union{Integer, AbstractVector{<:Integer}}) =
```

Comment:
I'd call this `parentcols` (we generally avoid "get" since it doesn't add anything).

Reply:
OK